### PR TITLE
020621

### PR DESCRIPTION
**Pull request by hvaranasi9**
Included subclass as discussed with Mateo. For some reason, the existing pulled version of Constants.java file in master base of the frc repository did not show the subclass. So, re-pulling.

### CODE REVIEW

**Comment by hvaranasi9** [comment]
Matteo,

Please view the comments that I wrote in the Robot.java file [in the autonomousPeriodic method] and provide advice accordingly. Thank you!

**Comment by Piphi5** [comment]
In order for you to use the Xbox Controller's bumpers, you need to initialize the Xbox Controller Object. This object contains methods for reading if the button was pressed. For reference, [here](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/XboxController.html) is a link to the Xbox Controller Documentation. 

Let me know if you have any other questions.

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -1,95 +1,103 @@
...
+  }
+
+  /** This function is called periodically during autonomous. */
+  @Override
+  public void autonomousPeriodic() { // is this the correct method to put this code in
+    if (leftBumper.isPressed) { // need to identify the left bumper event....
+      mMotor.setSpeed(Constants.IntakeConstants.kIntakePower);
+    } else if (rightBumper.isPressed) {
+      mMotor.setSpeed(-1 * Constants.IntakeConstants.kIntakePower);
+    }
```
Comment (Piphi5) [inline src/main/java/frc/robot/Robot.java]: This should be within teleopPeriodic

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -1,95 +1,103 @@
...
+ * each mode, as described in the TimedRobot documentation. If you change the name of this class or
+ * the package after creating this project, you must also update the build.gradle file in the
+ * project.
+ */
+public class Robot extends TimedRobot {
+  private Command m_autonomousCommand;
+
+  private RobotContainer m_robotContainer;
+
+  private VictorSPX mMotor = new VictorSPX(Constants.IntakeConstants.VictorSPXPort);
```
Comment (Piphi5) [inline src/main/java/frc/robot/Robot.java]: I would suggest a more descriptive name (e.g. mIntakeMotor)

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (Piphi5) [inline src/main/java/frc/robot/Robot.java]: Make sure you set the motor speed to zero when nothing is pressed

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -1,95 +1,103 @@
...
+    if (m_autonomousCommand != null) {
+      m_autonomousCommand.schedule();
+    }
+  }
+
+  /** This function is called periodically during autonomous. */
+  @Override
+  public void autonomousPeriodic() { // is this the correct method to put this code in
+    if (leftBumper.isPressed) { // need to identify the left bumper event....
+      mMotor.setSpeed(Constants.IntakeConstants.kIntakePower);
```
Comment (Piphi5) [inline src/main/java/frc/robot/Robot.java]: A VictorSPX does not have a setSpeed method. [Here ](https://www.ctr-electronics.com/downloads/api/java/html/classcom_1_1ctre_1_1phoenix_1_1motorcontrol_1_1can_1_1_victor_s_p_x.html) is the documentation regarding the VictorSPX

**Review by Piphi5 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -1,95 +1,103 @@
...
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
+
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+
```
Comment (Piphi5) [inline src/main/java/frc/robot/Robot.java:12]: Don't forget to import the XboxController and VIctorSPX code

**Pull request closed without merging**